### PR TITLE
Issue #2154: Add vagrant-nfs-fix role to fix NFS timeout issues on macOS

### PR DESCRIPTION
**Pull request by oxyc**
It's been a long time since I did anything ansible let alone init.d scripts. Therefore this needs reviews.

It's based on the firewall role, with the inital fix from @MokDevelopment and the systemd service by @susannecoates.

1. Is there an issue running it as root?
2. Can we somehow detect if the provisioning host is macos?
3. I'm just using `touch` since I'm guessing a timestamp modification should be enough? As I type this, I'm just realizing maybe I should add back the `rm` so the file doesnt accidentally get committed to git on the host.
4. I dont have diskspace to build any of the RHEL boxes :/ Could someone try it out?
5. I don't actually see the issue so someone should test it.

Co-authored-by: MokDevelopment
Co-authored-by: Susanne Coates

### CODE REVIEW

File: provisioning/roles/drupalvm.vagrant-nfs-fix/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+vagrant_nfs_fix_keepalive_file: "/vagrant/.vagrant-nfs-fix-keepalive.tmp"
```
Comment (geerlingguy) [inline provisioning/roles/drupalvm.vagrant-nfs-fix/defaults/main.yml:2]:
```suggestion
vagrant_nfs_fix_keepalive_file: "/vagrant/.vagrant-nfs-fix-keepalive.tmp"

```

File: provisioning/roles/drupalvm.vagrant-nfs-fix/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    mode: 0755
+
+- name: Copy Vagrant NFS fix init script into place.
+  template:
+    src: vagrant-nfs-fix.init.j2
+    dest: /etc/init.d/vagrant-nfs-fix
+    owner: root
+    group: root
+    mode: 0755
+  when: ansible_service_mgr != 'systemd'
```
Comment (geerlingguy) [inline provisioning/roles/drupalvm.vagrant-nfs-fix/tasks/main.yml:17]: I believe all supported OSes now include systemd, so we can drop the init script... right?

File: provisioning/roles/drupalvm.vagrant-nfs-fix/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    owner: root
+    group: root
+    mode: 0644
+  when: ansible_service_mgr == 'systemd'
+
+- name: Configure the NFS fix service.
+  service:
+    name: vagrant-nfs-fix
+    state: started
+    enabled: true
```
Comment (geerlingguy) [inline provisioning/roles/drupalvm.vagrant-nfs-fix/tasks/main.yml:32]:
```suggestion
    enabled: true

```

**Review by geerlingguy — CHANGES_REQUESTED** [review]
Looks fine, though I think we can ditch the init script as Drupal VM doesn't support any OS version old enough to not have systemd anymore.

File: provisioning/roles/drupalvm.vagrant-nfs-fix/templates/vagrant-nfs-fix.unit.j2 [hunk 4]
```diff
@@ -9,4 +9,4 @@ RestartSec=1
 ExecStart=/usr/sbin/vagrant-nfs-fix
 
 [Install]
-WantedBy=multi-user.target
\ No newline at end of file
+WantedBy=multi-user.target
```
Comment (oxyc) [inline provisioning/roles/drupalvm.vagrant-nfs-fix/templates/vagrant-nfs-fix.unit.j2:12]: @geerlingguy are this fine or do you want the actual empty line at the end?

File: default.config.yml [hunk 5]
```diff
@@ -42,6 +42,10 @@ vagrant_plugins:
   - name: vagrant-vbguest
   - name: vagrant-hostsupdater
 
+# Automatically fix timeout issues with NFS on macOS. See
+# https://github.com/geerlingguy/drupal-vm/issues/2154
+vagrant_nfs_fix_enabled: "{{ vagrant_synced_folder_default_type == 'nfs' }}"
```
Comment (oxyc) [inline default.config.yml:47]: Still not sure if this should really run by default just because it's nfs. Maybe it should be opt-in?

Comment (susannecoates) [reply default.config.yml:47]: Since it seems to only be a problem on Big Sur, maybe it should be opt in?

Comment (geerlingguy) [reply default.config.yml:47]: Yeah... maybe just set it false and we can add a note in the docs under this page? http://docs.drupalvm.com/en/latest/getting-started/installation-macos/ like:

## Big Sur

If you are using NFS shares on macOS Big Sur, and you have problems with the NFS mount going away, please set `vagrant_nfs_fix_enabled: true` in your `config.yml`.

Something like that.

Comment (susannecoates) [reply default.config.yml:47]: Yep, sounds good!

**Comment by oxyc** [comment]
While I have you, it would be much appreciated if you could give https://github.com/geerlingguy/packer-drupal-vm/issues/5 another try btw! It built successfully for me back when I tried at least.

**Review by geerlingguy — APPROVED** [review]

**Comment by marcelovani** [comment]
Had the same issue after upgrading to BigSur. Thanks for the hack to fix it. I wonder how lot its going to take for Vagrant to do a proper fix.

**Pull request merged**